DynamicTablesPkg/Library/Common/AmlLib/CodeGen/AmlCodeGen.c

```diff
@@ -4351,6 +4363,18 @@ AmlCreatePsdNode (
     goto error_handler;
   }
 
+  Status = AmlVarListAddTail (
+             (AML_NODE_HANDLE)PsdParentPackage,
+             (AML_NODE_HANDLE)PsdPackage
+             );
+  if (EFI_ERROR (Status)) {
+    ASSERT_EFI_ERROR (Status);
+    FreePool (IntegerNode);

```

Comment:
I think this line should be deleted as IntegerNode is already attached.
Otherwise the patch looks good to me on worked on my setup

Reply:
Yup copy paste error, will fix